packages/database/src/object-list.ts

```diff
@@ -1,56 +1,79 @@
-'use server';
-
-import {objectLists, insertObjectItemSchema, objectItems} from './db/schema';
+import {
+  objectLists,
+  objectItems,
+  insertObjectItemSchema,
+  insertObjectListSchema,
+} from './db/schema';
+import {InferSelectModel, sql} from 'drizzle-orm';
 import {db} from './db/connection';
 import {iriToHash} from './iri-to-hash';
+import {DBQueryConfig, eq} from 'drizzle-orm';
 
-async function getListsByCommunityId(communityId: string) {
-  return db.query.objectLists.findMany({
-    where: (objectLists, {eq}) => eq(objectLists.communityId, communityId),
-  });
+interface Option {
+  withObjects?: boolean;
+  limitObjects?: number;
 }
 
-async function getCommunityListsWithObjects(communityId: string) {
+interface ObjectList extends InferSelectModel<typeof objectLists> {
+  objects?: InferSelectModel<typeof objectItems>[];
+}
+
+export async function getByCommunityId(
+  communityId: string,
+  {withObjects, limitObjects}: Option = {withObjects: false}
+): Promise<ObjectList[]> {
+  const options: DBQueryConfig = {};
+
+  if (withObjects) {
+    options.with = {
+      objects: limitObjects ? {limit: limitObjects} : true,
+    };
+  }
+
   return db.query.objectLists.findMany({
+    ...options,
     where: (objectLists, {eq}) => eq(objectLists.communityId, communityId),
-    with: {
-      objects: true,
-    },
   });
 }
 
-interface CreateListForCommunityProps {
+export async function countByCommunityId(communityId: string) {
+  const result = await db
+    .select({count: sql<number>`count(*)`})
+    .from(objectLists)
+    .where(eq(objectLists.communityId, communityId));
+  return result[0].count;
```

Comment:
To be sure: is there a scenario where index 0 of `result` does not exist (i.e. should we check for its existence)?

Reply:
I don't think there is a scenario. The Drizzle orm aggregation functions always return an array with one value.

Reply:
Perhaps add a comment explaining this (i.e. we assume that a third-party library always returns an array with one value that is an object with the `count` prop)?